db/column_family.cc

```diff
@@ -342,6 +342,13 @@ ColumnFamilyOptions SanitizeOptions(const ImmutableDBOptions& db_options,
     result.max_compaction_bytes = result.target_file_size_base * 25;
   }
 
+  if (result.compaction_style == kCompactionStyleLevel &&
```

Comment:
If we move this check, what will break? If nothing will break, I would just remove this check.

Reply:
I don't think removing this will break anything, but there is no compaction trigger based on `periodic_compaction_seconds` in other compaction styles. So, it is just that we will end up computing the files (specifically `files_marked_for_periodic_compactions_`) for no reason ... it would just be wasted work for now. Do you still prefer to remove the check?

Reply:
It's up to you then.